linfa-svm/src/lib.rs

```diff
@@ -278,3 +278,52 @@ impl<'a, A: Float, T> fmt::Display for Svm<'a, A, T> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::Svm;
+    use linfa::dataset::Dataset;
+    use linfa::prelude::*;
+    use linfa_kernel::{Kernel, KernelMethod};
+    use ndarray::Array1;
+    #[test]
+    fn test_iter_folding_for_classification() {
+        let mut dataset = linfa_datasets::winequality().map_targets(|x| *x > 6);
+        let params = Svm::params().pos_neg_weights(7., 0.6);
+
+        let avg_acc = dataset
+            .iter_fold::<Kernel<_>, _, _, _>(4, &params, |svm_params, training_set| {
+                let train_kernel = Kernel::params()
+                    .method(KernelMethod::Gaussian(80.0))
+                    .transform(&training_set);
+                svm_params.fit(&train_kernel)
+            })
+            .map(|(model, valid)| {
+                model
+                    .predict(&valid)
+                    .map_targets(|x| **x > 0.0)
+                    .confusion_matrix(&valid)
+                    .accuracy()
+            })
+            .sum::<f32>()
+            / 4_f32;
+        assert!(avg_acc >= 0.5)
+    }
```

Comment:
This is how `iter_fold` looks for classification using `Svm`.  The ugly part is that it is unable to determine the type of the `Record` object used by the fittable model by itself, so I'm forced to write it down explicitly. 

The only cloning happening here is during the call to `fit` where the kernel is cloned inside the SVM